### ISSUE TITLE
VimFx 0.25.0 doesn't work on Firefox Nightly 115.0a1

### ISSUE THREAD

**Issue by Ace-Who** [akhodakivskiy/VimFx#993]
Firefox Nightly 115.0a1, Windows 10 1903 x64

It was working fine on Nightly earlier in this year before I switched to Firefox Stable and stopped using Nightly. Now I'm running Nightly again but VimFx doesn't work. Its icon doesn't appear on toolbar and the page chrome://vimfx/content/options.xhtml is blank.

LegacyFox files are still present in the program directory:

```
PS> ls -Recurse defaults,legacy,config.js,legacy.manifest,Makefile,LICENSE,README


    目录: C:\Program Files\Firefox Nightly\defaults


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        2023/5/30     07:48                pref


    目录: C:\Program Files\Firefox Nightly\defaults\pref


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       2022/10/15     07:50            429 channel-prefs.js
-a----        2022/4/29     22:15            134 config-prefs.js


    目录: C:\Program Files\Firefox Nightly\legacy


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        2022/4/29     22:15          11811 BootstrapLoader.jsm
-a----        2022/4/29     22:15          51603 RDFDataSource.jsm
-a----        2022/4/29     22:15           3182 RDFManifestConverter.jsm


    目录: C:\Program Files\Firefox Nightly


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        2022/4/29     22:15            690 config.js
-a----        2022/4/29     22:15             24 legacy.manifest
-a----        2022/4/29     22:15            497 Makefile
-a----        2022/4/29     22:15          16726 LICENSE
-a----        2022/4/29     22:15           1077 README
```

Browser Console:

![图片](https://github.com/akhodakivskiy/VimFx/assets/17397070/2121772f-0c99-49e9-b20e-6253060e80d8)

Restart:

![图片](https://github.com/akhodakivskiy/VimFx/assets/17397070/3bd39bd5-428f-494a-bc10-47b5f3914f72)

**Comment by girst** [comment]
On Tue, May 30, 2023 at 09:09:20AM -0700, Ace-Who wrote:
>Closed #993 as completed.

please don't close tickets that aren't fixed.

**Comment by ZWindL** [comment]
> this is due to a change implemented in mozilla115, i have a patch ready but not tested yet: https://git.gir.st/VimFx.git/commitdiff/6129b719059cc970aa7f19c9509528204a7c1370 if you want to help, i'd appreciate it if you could try this one out. i hope i can find some time this week/weekend to push a new version. i don't like that this increases the minimal supported firefox version by quite a bit, but can't help it.

I tried this patch on my laptop, it works fine! Thank you!

**Comment by girst** [comment]
fixed in v0.26.0. vimfx now requires at least firefox 84.

**Comment by Ace-Who** [comment]
Thanks. I created another issue about v0.26.0.
